src/include/kompute/Shader.hpp

```diff
@@ -46,13 +48,123 @@ class Shader {
      * @param source An individual raw glsl shader in string format
      * @param entryPoint The function name to use as entry point
      * @param definitions List of pairs containing key value definitions
+     * @param resourcesLimit A list that contains the resource limits for the GLSL compiler
      * @return The compiled SPIR-V binary in unsigned int32 format
      */
     static std::vector<uint32_t> compile_source(
             const std::string& source,
             const std::string& entryPoint = "main",
-            std::vector<std::pair<std::string,std::string>> definitions = {});
+            std::vector<std::pair<std::string,std::string>> definitions = {},
+            const TBuiltInResource resources = defaultResource);
 
+private:
+    // The default resource limit for the GLSL compiler, can be overwritten
```

Comment:
is everythign else except max-lights the same as the standalone file? May be worth adding a comment also linking to the github file

Reply:
Yeah I was about to write an explanation for it. It's actually not complete, as I don't know which ones are exactly required and what it's not required in our usage, like I'm sure we won't need `MaxLights` or 'MaxFragmentUniformComponents' so I set them to zero, but  I honestly don't know about `MaxCombinedShaderOutputResources`. Do you know what exactly we need?

> May be worth adding a comment also linking to the github file

Yeah sure